homeassistant/components/pvpc_hourly_pricing/__init__.py

```diff
@@ -119,4 +129,43 @@ async def async_update_options(hass: HomeAssistant, entry: ConfigEntry) -> None:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
+
+class ElecPricesDataUpdateCoordinator(DataUpdateCoordinator[Mapping[datetime, float]]):
+    """Class to manage fetching Electricity prices data from API."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize."""
+        self.api = PVPCData(
+            session=async_get_clientsession(hass),
+            tariff=entry.data[ATTR_TARIFF],
+            local_timezone=hass.config.time_zone,
+            power=entry.data[ATTR_POWER],
+            power_valley=entry.data[ATTR_POWER_P3],
+        )
+        super().__init__(
+            hass, _LOGGER, name=DOMAIN, update_interval=timedelta(minutes=30)
+        )
+        self._entry = entry
+
+    @property
+    def entry_id(self) -> str:
+        """Return entry ID."""
+        return self._entry.entry_id
+
+    async def _async_update_data(self) -> Mapping[datetime, float]:
+        """Update electricity prices from the ESIOS API."""
+        now = dt_util.utcnow()
+        try:
+            prices = await self.api.async_update_prices(now)
+            if not prices:
+                raise UpdateFailed
+
+            self.api.process_state_and_attributes(now)
```

Comment:
Can this call raise? Otherwise please move it out of the try... except block so we only wrap the line that can raise.